### PR TITLE
Harden Show-FailureLogs.ps1 against failures where the files are not present between Get-ChildItem and the output of the file

### PR DESCRIPTION
**Pull request by danieljurek**
# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/main/CONTRIBUTING.md#pull-requests).

- [ ] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [ ] Doxygen docs
- [ ] Unit tests
- [ ] No unwanted commits/changes
- [ ] Descriptive title/description
  - [ ] PR is single purpose
  - [ ] Related issue listed
- [ ] Comments in source
- [ ] No typos
- [ ] Update changelog
- [ ] Not work-in-progress
- [ ] External references or docs updated
- [ ] Self review of PR done
- [ ] Any breaking changes?

### CODE REVIEW

File: eng/scripts/Show-FailureLogs.ps1 [hunk 1]
```diff
@@ -19,7 +19,11 @@ foreach ($logFile in $filteredLogs)
     Write-Host "=============================================================================================================================="
     Write-Host "Log file: $logFile"
     Write-Host "=============================================================================================================================="
-    Get-Content $logFile | Write-Host
+    try { 
+        Get-Content $logFile | Write-Host
```
Comment (weshaggard) [inline eng/scripts/Show-FailureLogs.ps1:23]: Why not do a Test-Path here instead of a try/catch?

Comment (weshaggard) [reply eng/scripts/Show-FailureLogs.ps1:23]: Also what is actually deleting the files in parallel here that would cause this to fail?

Comment (danieljurek) [reply eng/scripts/Show-FailureLogs.ps1:23]: If it's a race condition we could still fail when we `Get-Content`

Comment (weshaggard) [reply eng/scripts/Show-FailureLogs.ps1:23]: Do you have any idea what would delete the file while we are looping over them?

**Review by antkmsft — APPROVED** [review]

**Comment by ahsonkhan** [comment]
Fixes https://github.com/Azure/azure-sdk-for-cpp/issues/5426

File: eng/scripts/Show-FailureLogs.ps1 (hunk 1 above)
Comment (weshaggard) [reply eng/scripts/Show-FailureLogs.ps1:23]: After having a look my best guess is that an empty log file is still around when you query for logs but then once the full vcpkg process shuts down it removes that empty file. Given that this is just about dumping the log file if it exists I'm find with this solution.

**Review by ahsonkhan — APPROVED** [review]

File: eng/scripts/Show-FailureLogs.ps1 [hunk 2]
```diff
@@ -19,7 +19,11 @@ foreach ($logFile in $filteredLogs)
...
+    } catch { 
+        Write-Host "Could not locate file found using Get-ChildItem $logFile"
+    }
 }
 
 exit 0
```
Comment (gearama) [inline eng/scripts/Show-FailureLogs.ps1:29]: new line :)

**Review by gearama — APPROVED** [review]

File: eng/scripts/Show-FailureLogs.ps1 (hunk 1 above)
Comment (danieljurek) [reply eng/scripts/Show-FailureLogs.ps1:25]: As far as I can tell, vcpkg should have exited by the time this script is running. 

I checked locally to see if the output of `Get-ChildItem` fed into `Get-Content` was somehow getting the wrong set of folders leading up to the file. That doesn't appear to be the case. I added more logging in the event that there's a factor on the build machine that's not being considered.

**Pull request merged**